app/models/evaluation.rb

```diff
@@ -29,7 +29,17 @@ def self.key_attributes
 
   def self.create_if_needed_and_update(key_attrs, other_attrs)
     if other_attrs[:instructor].is_a?(String)
-      other_attrs[:instructor_id] = Instructor.where(name: Instructor.normalize_name(other_attrs[:instructor])).first_or_create.id
+      lst_name = other_attrs[:instructor].split.first
+      fst_name = other_attrs[:instructor].split.last
+      search_str1 = lst_name + "%" + " " + fst_name + "%"
+      search_str2 = fst_name + "%" + " " + lst_name + "%"
+      if Instructor.where('name like ?', search_str1).length != 0
```

Comment:
I think `.count > 0` is a more intuitive method to use rather than `.length != 0`